### PR TITLE
Feature/admin auth

### PR DESCRIPTION
**Pull request by jochen42**
implements #12

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -20,6 +20,15 @@ provider "kong" {
 }
 ```
 
+Optionaly you can configure Username and Password for BasicAuth:
+```hcl
+provider "kong" {
+    kong_admin_uri  = "http://myKong:8001"
+    kong_admin_user = "youruser"
```
Comment (kevholditch) [inline README.md]: update variable to `kong_admin_username` please

File: README.md [hunk 2]
```diff
@@ -20,6 +20,15 @@ provider "kong" {
...
+    kong_admin_pass = "yourpass"
```
Comment (kevholditch) [inline README.md]: update variable to `kong_admin_password` please

File: kong/provider.go [hunk 3]
```diff
@@ -16,6 +16,18 @@ func Provider() terraform.ResourceProvider {
 				DefaultFunc: envDefaultFuncWithDefault("KONG_ADMIN_ADDR", "http://localhost:8001"),
 				Description: "The address of the kong admin url e.g. http://localhost:8001",
 			},
+			"kong_admin_user": &schema.Schema{
```
Comment (kevholditch) [inline kong/provider.go]: update to `kong_admin_username`

File: kong/provider.go [hunk 4]
```diff
@@ -16,6 +16,18 @@ func Provider() terraform.ResourceProvider {
...
+				Type:        schema.TypeString,
+				Optional:    true,
+				DefaultFunc: envDefaultFuncWithDefault("KONG_ADMIN_USER", ""),
+				Description: "An basic auth user for kong admin",
+			},
+			"kong_admin_pass": &schema.Schema{
```
Comment (kevholditch) [inline kong/provider.go]: Update to `kong_admin_password`

**Comment by kevholditch** [comment]
The build is still failing can you get the tests to pass please

**Comment by jochen42** [comment]
yes. that's wired. if i run the test local with 'go test -v ./kong -run="TestAcc"' it passes. but i will figure out...

**Comment by jochen42** [comment]
so the build is green nw by manual checking nil value. im am new in go and terraform, but i think the "envDefaultFuncWithDefault" method should ensure an empty string. but perhaps "Optional" does something unexpected.

File: kong/provider.go [hunk 5]
```diff
@@ -54,8 +66,18 @@ func envDefaultFuncWithDefault(key string, defaultValue string) schema.SchemaDef
 }
 
 func providerConfigure(d *schema.ResourceData) (interface{}, error) {
+	Username := ""
```
Comment (kevholditch) [inline kong/provider.go]: this variable should start with a lowercase letter so `username`

File: kong/provider.go [hunk 6]
```diff
@@ -54,8 +66,18 @@ func envDefaultFuncWithDefault(key string, defaultValue string) schema.SchemaDef
...
+	if d.Get("kong_admin_username") != nil {
+		Username = d.Get("kong_admin_username").(string)
+	}
+	Password := ""
```
Comment (kevholditch) [inline kong/provider.go]: variable should start with a lowercase letter `password`

**Pull request merged**